lib/processors/execution_hook_processors.rb

```diff
@@ -71,6 +72,7 @@ def process_step_execution_start_request(message)
 
     def process_step_execution_end_request(message)
       response = handle_hooks_execution(MethodCache.get_after_step_hooks, message,message.stepExecutionEndingRequest.currentExecutionInfo)
+      response.executionStatusResponse.executionResult.screenShot += Gauge::GaugeScreenshot.instance.pending_screenshot
```

Comment:
i know I said we only need it for `step` ending. But turns out we need it for other hooks as well, just like messages. see https://github.com/getgauge/gauge-csharp/issues/142